task/backend/analytical_storage.go

```diff
@@ -152,18 +153,45 @@ func (as *AnalyticalStorage) FindRuns(ctx context.Context, filter influxdb.RunFi
 		filterPart = fmt.Sprintf(`|> filter(fn: (r) => r.runID > %q)`, filter.After.String())
 	}
 
+	parsedAfterTime := time.Time{}
+	parsedBeforeTime := time.Now()
+	constructedTimeFilter := ""
+	if feature.TimeFilterFlags().Enabled(ctx) {
+		if filter.AfterTime != "" {
+			tmpParsedAfter, err := time.Parse(time.RFC3339, filter.AfterTime)
+			if err != nil {
+				return nil, 0, err
```

Comment:
it may be beneficial to add some context to this error to help the user differentiate between this and the before time parsing
```suggestion
				return nil, 0, fmt.Errorf("failed parsing after time: %s", err.Error())
```